youtube_dl/extractor/bandlab.py

```diff
@@ -0,0 +1,172 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+
+from ..compat import (
+        compat_str,
+)
+
+from ..utils import (
+    strip_or_none,
+    try_get,
+    url_or_none,
+)
+
+
+class BandlabIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?bandlab\.com/post/(?P<id>[^/]+)'
+
+    _TESTS = [{
+        'url': 'https://www.bandlab.com/post/f5f04998635a44ea819cacdba7ae2076_f8d8574c3bdaec11b6562818783151a1',
+        'info_dict': {
+            'id': 'f5f04998635a44ea819cacdba7ae2076_f8d8574c3bdaec11b6562818783151a1',
+            'ext': 'm4a',
+            'title': 'ON MY OWN (unreleased)',
+            'artist': 'Michael MacDonald',
+        },
+    }]
+
+    def _real_extract(self, url):
+        track_id = self._match_id(url)
+        config = self._download_json(
+            'http://www.bandlab.com/api/v1.3/posts/%s' % track_id, track_id)
+        track_url = config['track']['sample']['audioUrl']
+
+        return {
+            'id': track_id,
+            'title': config['track']['name'],
+            'url': track_url,
+            'artist': config['creator']['name']
+        }
+
+
+class BandlabAlbumOrPlaylistIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?bandlab\.com/[^/]+/(?P<kind>albums|collections)/(?P<id>[^/]+)'
+    _TRACK_ID_RE = re.compile(r'.+/(?P<id>[^/]+)\.m4a')
```

Comment:
It's not really necessary to do this because Python's `re` module caches compiled REs.
```suggestion
    _TRACK_ID_RE = r'.+/(?P<id>[^/]+)\.m4a'
```